### DISCUSSION TITLE
Dynamic extensions: [params].[extension].ts

### DISCUSSION THREAD

**Discussion by reesericci** [withastro/roadmap#283] [Proposal]
Hello!

I think it would be useful if Astro supported dynamic extensions. I was creating an API that returns a string which is to be turned into a file to be put on a route. The only thing is, that file extension might change and it is specified in the JSON coming back from the API. In this case it's possible to not serve the file over the API and hard-code it, but that's not always the case.

Thanks!

--reese

**Comment by andygrunwald** [comment]
I have a similar usecase.
I use Astro to render HTML files. However, my goal is to render the same data as an XML file.
The current file is `something.astro`, which renders as `/something/index.html`.
I aim to also render `something.xml.astro`, which renders to `something.xml`.

**Comment by florian-lefebvre** [comment]
This is possible with a setup like this:

```ts
// /src/pages/[...slug].ts
export const getStaticPaths = () => {
    return [{ params: { slug: 'foo' }}, { params: { slug: 'bar.xml' }}]
}

export const GET = ({ params }) => {
    return new Response(params.slug)
}
```
